mitmproxy/addons/upload.py

```diff
@@ -0,0 +1,124 @@
+import os.path
+import typing
+import random
+import datetime
+import time
+from urllib.request import Request, urlopen
+from urllib.error import HTTPError, URLError
+import mimetypes
+
+from mitmproxy import command
+from mitmproxy import exceptions
+from mitmproxy import flowfilter
+from mitmproxy import io
+from mitmproxy import ctx
+from mitmproxy import flow
+
+import mitmproxy.types
+
+from mitmproxy.addons import multipart
+
+
+class Upload:
+    def __init__(self):
+        self.stream = None
+        self.filt = None
+        self.active_flows = set()  # type: Set[flow.Flow]
```

Comment:
I don't understand why all this is here. The `share` command accepts a list of flows which should be uploaded, everything else here is not necessary.

Reply:
I am not able to understand mitmproxy/addons/save.py works. Since I did not want to break anything, I did not change any of the methods. for example I couldn't figure out why tcp_start and tcp_end methods are there? Can you please give little explaination on this

Reply:
The save addon can be configured to continually stream all requests to a file. In this case we use the tcp_start/tcp_end hooks to register when a flow has been finished.

Reply:
Can you tell me how it can be configured.